packages/neo-one-client/src/user/LocalUserAccountProvider.ts

```diff
@@ -823,6 +841,172 @@ export class LocalUserAccountProvider<TKeyStore extends KeyStore, TProvider exte
     );
   }
 
+  private async consolidate({
+    inputs,
+    transactionUnsignedIn,
+    from,
+    monitor,
+    byteLimit,
+  }: {
+    readonly transactionUnsignedIn: TransactionModel;
+    readonly inputs: ReadonlyArray<UnspentOutput>;
+    readonly from: UserAccountID;
+    readonly monitor?: Monitor;
+    readonly byteLimit: number;
+  }): Promise<TransactionModel> {
+    const newInputSize = 68;
+    const newOutputSize = 120;
+    const messageSize = transactionUnsignedIn.serializeUnsigned().toString('hex').length;
+
+    if (byteLimit - messageSize < 0) {
+      throw new TransactionSizeExceedsLimitError(messageSize, byteLimit);
+    }
+    if (byteLimit - messageSize - newInputSize < 0) {
+      return transactionUnsignedIn;
+    }
+
+    const inputMap = (): { [key: string]: ReadonlyArray<UnspentOutput> } => ({});
+    const inputArray = (): ReadonlyArray<InputModel> => [];
+    const outputArray = (): ReadonlyArray<OutputModel> => [];
+    const getMessageSize = (numInputs: number, numOutputs: number) =>
```

Comment:
It's generally a good idea to use an object as an input when you have multiple primitive arguments. As it stands, there's nothing to stop you from mixing up `numInputs` and `numOutputs`, whereas if the input was `{ numInputs: number; numOutputs: number }`, it'd be much more difficult to make a mistake. Also much clearer for the reader that you're passing the right arguments.

Reply:
Also, this should probably be `numNewInputs` and `numNewOutputs` - it's not the total number of inputs/outputs of the transaction.